spring-integration-kafka/src/main/java/org/springframework/integration/kafka/inbound/KafkaInboundEndpoint.java

```diff
@@ -31,7 +33,7 @@
  * @since 6.0
  *
  */
-public interface KafkaInboundEndpoint {
+public interface KafkaInboundEndpoint<K, V> {
```

Comment:
This would be a breaking change for the version we are going, but at the same time the `doWithRetry()` contract change to the `ConsumerRecord` is a breaking change anyway.
So, yeah, probably let's stick with the current version where we can introduce breaking change, but let's get rid of these generics! The `doWithRetry()` doesn't care what are those key and value types.
So, simple `ConsumerRecord<?, ?>` as a param should be enough. Similar to what we have so far with a `consumer` param.

Reply:
Yes, it's done.